### ISSUE TITLE
[BUG] GM_xmlhttpRequest fails if given a URL object (interoperability)

### ISSUE THREAD

**Issue by ppixiv** [violentmonkey/violentmonkey#1578]
Calling GM_xmlhttpRequest with the url field set to a URL object throws an exception:

```
VM69 injected.js:1 Uncaught TypeError: Cannot read properties of null (reading 'node')
    at VM69 injected.js:1:4687
```

Neither onload nor onerror are called, so the call never completes.  Simple to work around, but it does work in TamperMonkey, so this worked fine for a long time in my code until a VM user ran into this.

Repro attached.  Load on https://ppixiv.github.com/ (or wherever you like).  When given a string it completes normally, but it fails internally when given a URL.  Tested in VM 2.13.1 with Chrome 104.0.5112.102.  [url_test.user.js.txt](https://github.com/violentmonkey/violentmonkey/files/9613927/url.test.user.js.txt)

**Comment by tophf** [comment]
I was about to indicate that this is the correct behavior because this parameter is described as string in [specification of XMLHttpRequest](https://xhr.spec.whatwg.org/#xmlhttprequest), in greasemonkey wiki, as well as ours. But then I saw that USVString/DOMString [calls ToString on the input](https://webidl.spec.whatwg.org/#es-USVString), which means it'd reasonable for us to do the same.

**Comment by tophf** [comment]
Fixed in 6189e4a0.
